src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -0,0 +1,623 @@
+                }
+                else
+                {
+                    // add extra line at end to end doc comment
+                    // also skip first newline and replace with non-newline
+                    newClassDocComment = SyntaxFactory.DocumentationCommentTrivia(
+                        SyntaxKind.MultiLineDocumentationCommentTrivia,
+                        SyntaxFactory.List(propertyParamComments.Skip(1)
+                            .Prepend(SyntaxFactory.XmlText(SyntaxFactory.XmlTextLiteral(" ").WithLeadingTrivia(exteriorTrivia)))))
+                        .WithAppendedTrailingTrivia(SyntaxFactory.ElasticCarriageReturnLineFeed);
+                }
+            }
+
+            var lastComment = classTrivia.LastOrDefault(trivia => trivia.IsRegularOrDocComment());
+            if (classDocComment == null || lastComment == classDocComment)
+            {
+                // doc comment was last non-whitespace/newline trivia or there was no class doc comment originally
+                return SyntaxFactory.TriviaList(classTrivia
+                    .Where(trivia => !trivia.IsDocComment())
+                    .Concat(propertyNonDocComments)
+                    .Append(SyntaxFactory.Trivia(newClassDocComment))
+                    .Select(trivia => trivia.AsElastic()));
+            }
+            else
+            {
+                // there were comments after doc comment
+                return SyntaxFactory.TriviaList(classTrivia
+                    .Replace(classDocComment.Value, SyntaxFactory.Trivia(newClassDocComment))
+                    .Concat(propertyNonDocComments)
+                    .Select(trivia => trivia.AsElastic()));
+            }
+        }
+
+        private static SyntaxTriviaList? GetExteriorTrivia(SyntaxNode declaration)
+        {
+            var potentialDocComment = declaration.GetLeadingTrivia().FirstOrNull(trivia => trivia.IsDocComment());
+
+            if (potentialDocComment?.GetStructure() is DocumentationCommentTriviaSyntax docComment)
+            {
+                // if single line, we return a normal single line trivia, we can format it fine later
+                if (docComment.IsKind(SyntaxKind.SingleLineDocumentationCommentTrivia))
+                {
+                    // first token of comment should have correct trivia
+                    return docComment.GetLeadingTrivia();
+                }
+                else
+                {
+                    // for multiline comments, the continuation trivia (usually "*") doesn't get formatted correctly
+                    // so we want to keep whitespace alignment across the entire comment
+                    return SearchInNodes(docComment.Content);
+                }
+            }
+            return null;
+
+            // helper recursion to find the first exterior trivia of the element that is after a newline token 
```

Comment:
> helper recursion

I don't think that means what you want it to means.

Reply:
I'd like to dive into elements to catch a case such as 
```
/** <summary> <b>
 *  comment
 * </b>
 * </summary>
 */ 
```
and find a ` *` exterior trivia, since the exterior trivia is only attached to text elements. So, when we see an xml element, we recurse into it to hopefully find text that has a newline from there

Reply:
I just meant the wording of "helper recursion" seemed a bit odd. "Helper method" or "Recursive method" would make more sense to me :)